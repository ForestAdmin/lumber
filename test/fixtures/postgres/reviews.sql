DROP TABLE IF EXISTS reviews;
CREATE TABLE reviews (
  id serial,
  book_id int NOT NULL,
  user_id int NOT NULL,
  review_content VARCHAR(255),
  rating int,
  published_date timestamp DEFAULT CURRENT_TIMESTAMP,
  PRIMARY KEY (id),
  FOREIGN KEY (book_id) REFERENCES books(id) ON DELETE CASCADE,
  FOREIGN KEY (user_id) REFERENCES users(id) ON DELETE CASCADE
);