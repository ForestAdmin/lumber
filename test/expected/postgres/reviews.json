{
  "fields":[
    {
      "name":"reviewContent",
      "nameColumn":"review_content",
      "type":"STRING",
      "primaryKey":false,
      "defaultValue":null
    },
    {
      "name":"rating",
      "nameColumn":"rating",
      "type":"INTEGER",
      "primaryKey":false,
      "defaultValue":null
    },
    {
      "name":"publishedDate",
      "nameColumn":"published_date",
      "type":"DATE",
      "primaryKey":false,
      "defaultValue":"now()"
    }
  ],
  "references":[
    {
      "isBelongsTo":true,
      "ref":"books",
      "foreignKey":"book_id",
      "foreignKeyName":"bookId",
      "as":"book",
      "association":"belongsTo"
    },
    {
      "isBelongsTo":true,
      "ref":"users",
      "foreignKey":"user_id",
      "foreignKeyName":"userId",
      "as":"user",
      "association":"belongsTo"
    }
  ],
  "primaryKeys":["id"],
  "options":{
    "underscored":true,
    "timestamps":false,
    "hasIdColumn":true,
    "hasPrimaryKeys":true
  }
}